.github/workflows/icu_valgrind.yml

```diff
@@ -16,22 +16,22 @@ on:
 
 jobs:
   clang-valgrind-test:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
     steps:
       - name: Install valgrind
         run: |
           set -ex;
           sudo apt-get -y update;
           sudo apt-get install -y valgrind;
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       - name: Build with debug enable
         env:
           CC: clang
           CXX: clang++
         run: |
           cd icu4c/source;
-          ./runConfigureICU --enable-debug Linux --disable-renaming && make -j tests;
+          CFLAGS=-gdwarf-4 CXXFLAGS=-gdwarf-4 ./runConfigureICU --enable-debug Linux --disable-renaming && make -j tests;
```

Comment:
I think what you need to do is put the CFLAGS and CXXFLAGS in the env section above next to the CC and CXX.